### PR TITLE
Switch from `-sUSE_PTHREAD` to the standard `-pthread`. NFC

### PR DESCRIPTION
**Pull request by sbc100**
We already supported this flag, but this change mostly updates all or tests and documentation to refer exclusively to that latter.

### CODE REVIEW

**Review by dschuff — APPROVED** [review]
:tada: this is gret
(it is `-pthread` and not `-pthreads` that we recommend, right? I can never remember which of those is the "correct" one)

File: test/test_core.py [hunk 1]
```diff
@@ -6878,7 +6877,7 @@ def test_freetype(self):
   })
   def test_sqlite(self, use_pthreads):
     if use_pthreads:
-      self.set_setting('USE_PTHREADS')
+      self.emcc_cflags.append('-pthread')
```
Comment (kripken) [inline test/test_core.py:6880]: What's the difference between `emcc_cflags` and `emcc_args`?

Comment (sbc100) [reply test/test_core.py:6880]: Oh good catch.. emcc_cflags doesn't exist :)

**Comment by sbc100** [comment]
> tada this is great (it is `-pthread` and not `-pthreads` that we recommend, right? I can never remember which of those is the "correct" one)

There is no `s` on the end.. `-pthreads` with an s is just an error.

Or at least I thought it was.. looks like it is for gcc..

```
$ gcc -pthreads ~/test/hello.c
gcc: error: unrecognized command-line option ‘-pthreads’; did you mean ‘-pthread’?
```

But maybe not for clang.. thats annoying,.

```
$ clang -pthreads ~/test/hello.c
```

**Comment by sbc100** [comment]
I guess we could followup to start accepting `-pthreads` (plural) like clang does?

**Comment by dschuff** [comment]
Honestly I could take or leave supporting `-pthreads` (unless `-pthreads` is more common/popular). I think CMake checks for both.

**Comment by sbc100** [comment]
> Honestly I could take or leave supporting `-pthreads` (unless `-pthreads` is more common/popular). I think CMake checks for both.

I did a bunch of history spelunking to try to find out why clang differs from gcc here.. but I didn't find out yet.

Got as far back as 2010: https://github.com/llvm/llvm-project/commit/34d77dc49aa988af94372d97700fbd86b269d5cb

**Comment by sbc100** [comment]
Found the initial commit that added `pthreads`.. claims it was adding "gcc options" so I guess it was an accident of history?

**Pull request merged**